src/txn/txn.c

```diff
@@ -709,45 +718,42 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
 		 * than stable timestamp.
 		 */
 		WT_ERR(__wt_timestamp_validate(
-		    session, "commit", ts, &cval, false));
+		    session, "commit", ts, &cval, !prepare));
 		txn->commit_timestamp = ts;
 		__wt_txn_set_commit_timestamp(session);
+		if (!prepare)
+			txn->durable_timestamp = txn->commit_timestamp;
 	}
 
-	prepare = F_ISSET(txn, WT_TXN_PREPARE);
 	if (prepare && !F_ISSET(txn, WT_TXN_HAS_TS_COMMIT))
 		WT_ERR_MSG(session, EINVAL,
 		    "commit_timestamp is required for a prepared transaction");
 
-	/* Durable timestamp is required for a prepared transaction. */
-	if (prepare) {
-		WT_ERR(__wt_config_gets_def(
-		    session, cfg, "durable_timestamp", 0, &cval));
-		if (cval.len != 0) {
-			WT_ERR(__wt_txn_parse_timestamp(
-			    session, "durable", &ts, &cval));
-			WT_ERR(__wt_timestamp_validate(
-			    session, "durable", ts, &cval, true));
-			txn->durable_timestamp = ts;
-		} else
-			/*
-			 * If durable timestamp is not given, commit timestamp
-			 * will be considered as durable timestamp.
-			 * TODO : error if durable timestamp is not given.
-			 */
-			txn->durable_timestamp = txn->commit_timestamp;
-
-	} else
-		txn->durable_timestamp = txn->commit_timestamp;
+	/*
+	 * Durable timestamp is required for a prepared transaction.
+	 * If durable timestamp is not given, commit timestamp will be
+	 * considered as durable timestamp. We don't flag error if durable
+	 * timestamp is not specified for prepared transactions, but will flag
+	 * error if durable timestamp is specified for non-prepared
+	 * transactions.
+	 */
+	WT_ERR(__wt_config_gets_def(
+	    session, cfg, "durable_timestamp", 0, &cval));
+	if (cval.len != 0) {
+		if (!prepare)
+			WT_ERR_MSG(session, EINVAL,
+			    "durable_timestamp should not be given for "
+			    "non-prepared transaction");
 
-	/* Durable timestamp should be later than stable timestamp. */
-	if (cval.len != 0)
+		WT_ERR(__wt_txn_parse_timestamp(
+		    session, "durable", &ts, &cval));
+		/* Durable timestamp should be later than stable timestamp. */
 		WT_ERR(__wt_timestamp_validate(
-		    session, "durable", txn->durable_timestamp, &cval, true));
-
-	WT_ERR(__txn_commit_timestamp_validate(session));
+		    session, "durable", ts, &cval, true));
+		txn->durable_timestamp = ts;
+	}
 
-	/* TODO : assert durable_timestamp. */
+	WT_ERR(__txn_commit_timestamps_validate(session));
 
 	/*
 	 * The default sync setting is inherited from the connection, but can
```

Comment:
Not related to this file, but at the end of all the changes - I'm surprised not to find test changes or a new test. There are at least 4 assertion tests `test_assert0[1234].py` in the test suite. This should be added to existing tests or a new one created.

Reply:
I have added new test test_assert_durable_ts01.py